processing/src/main/java/io/druid/query/extraction/LookupExtractor.java

```diff
@@ -50,11 +78,36 @@
    * returning an empty list implies that user want to ignore such a lookup value.
    * In the other hand returning a list with the null element implies user want to map the none existing value to the key null.
    */
-  List<String> unApply(String value);
+
+  abstract List<String> unApply(String value);
+
+  /**
+   * @param values Iterable of values for which will perform reverse lookup
+   *
+   * @return Returns {@link Map} whose keys are the contents of {@code values} and whose values are computed on demand using the reverse lookup function {@link #unApply(String)}
+   * or empty map if {@code values} is `null`
+   * User can override this method if there is a better way to perform bulk reverse lookup
+   */
+
+  Map<String, List<String>> unApply(@NotNull Iterable<String> values)
```

Comment:
Can we also make this `unApplyAll`?

Also, a bit of a nut that I just noticed, I think I'd prefer `unapply` instead of `unApply`